src/components/portal/portal.tsx

```diff
@@ -11,10 +14,11 @@ export class Portal extends React.PureComponent<PortalProps> {
 
     public render() {
         const {children, ...rest} = this.props;
+        const id = getRootId(this);
         this.portalContent = (
-            <div {...rest} data-automation-id="PORTAL">{children}</div>
+            <div id={id} {...rest} data-automation-id="PORTAL">{children}</div>
         );
-        return null;
+        return <span data-id={id} data-automation-id="ID_SPAN"/>;
```

Comment:
should this span be styled like `display: none`?

Reply:
i think it will be too confusing - `data-automation-id` is the string we use to locate something in the DOM, should not have a value there to refer another node. 
we can totally implement it with some string processing but i think separating them like i did conforms to how we do things with automation-ids. @tobich your thoughts?

about the display none - not sure it matters since its going to be considered `absent` anyway, i think. will add the style just in case